### PR TITLE
Transfer Sergey's pybcl branch from the old private repo

### PR DESCRIPTION
**Pull request by roccomoretti**
@lyskov Made a number of changes for the pybcl branch in the RosettaCommons BCL repo, which is based off of the old, pre-MIT private repo. (These were merged with https://github.com/RosettaCommons/bcl/pull/14 )

This PR attempts to graft those changes back onto the public version of the repository.

### CODE REVIEW

**Comment by lyskov** [comment]
Thank you @roccomoretti for transferring it to public BCL repo!

File: include/graph/bcl_graph_common_subgraph_isomorphism.h [hunk 1]
```diff
@@ -654,9 +654,10 @@ namespace bcl
       //////////////////////////////////
 
         //! @brief default constructor ( optional in this case)
-        IntermediateSolution()
-        {
-        }
+        // commented since current implementation could not be compiled because member variables requiring explicit instantiation is no instantiated
+        // IntermediateSolution()
```
Comment (brownbp1) [inline include/graph/bcl_graph_common_subgraph_isomorphism.h:658]: Flagging this in case tests fail.

**Review by brownbp1 — APPROVED** [review]
Looks good. Build passes example checks, too. Thanks @roccomoretti for bringing over @lyskov changes.

**Comment by lyskov** [comment]
Awesome, - thank you for taking care of this @roccomoretti @brownbp1 !

**Pull request merged**